applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.hpp

```diff
@@ -0,0 +1,187 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors: Marjan Fathian
+//                Wijtze Pieter Kikstra
+//                Anne van de Graaf
+//
+
+#pragma once
+
+// System includes
+#include <cmath>
+#include <iostream>
+
+// Project includes
+#include "custom_elements/small_strain_U_Pw_diff_order_element.hpp"
+#include "custom_utilities/stress_strain_utilities.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "utilities/math_utils.h"
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+
+namespace Kratos
+{
+
+class ApplyCPhiReductionProcess : public Process
+{
+public:
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyCPhiReductionProcess);
+
+    ApplyCPhiReductionProcess(ModelPart& model_part, const Parameters&)
+        : Process(Flags()), mrModelPart(model_part)
+    {
+    }
+
+    ~ApplyCPhiReductionProcess() override = default;
+
+    void ExecuteInitializeSolutionStep() override
+    {
+        KRATOS_TRY
+
+        // Recognize whether a step is restarted. If so, try again with a smaller reduction increment.
+        if (mrModelPart.GetProcessInfo().GetValue(NUMBER_OF_CYCLES) > 1) mReductionIncrement *= 0.5;
```

Comment:
We might get rid of the comment if we would introduce a member function with a descriptive name, e.g.:
```
bool IsStepRestarted() const
{
    return mrModelPart.GetProcessInfo().GetValue(NUMBER_OF_CYCLES) > 1;
}
```
And then call it here:
```suggestion
        if (IsStepRestarted()) mReductionIncrement *= 0.5;
```

Reply:
Indeed this is clearer. Introduced.